lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -236,45 +253,52 @@ def self.upload_binary(app_id, binary_path)
           request.headers["Authorization"] = "Bearer " + auth_token
         end
       rescue Faraday::ResourceNotFound
-        UI.crash!(ErrorMessage::INVALID_APP_ID)
+        UI.crash!(ErrorMessage::INVALID_APP_ID + app_id)
       rescue Errno::ENOENT
-        UI.crash!(ErrorMessage::APK_NOT_FOUND)
+        UI.crash!(ErrorMessage::APK_NOT_FOUND + binary_path)
       end
 
+      # Uploads the binary
+      #
+      # Returns the id of the release. Only happens on a successful release, on a fail release a messsage notifies the user.
       def self.upload(app_id, binary_path)
         upload_token = get_upload_token(app_id, binary_path)
-        status = upload_status(app_id, upload_token)
-        if status == "SUCCESS"
+        upload_status_response = get_upload_status(app_id, upload_token)
+        if upload_status_response.success?
           UI.message("This APK/IPA has been uploaded before. Skipping upload step.")
         else
           UI.message("This APK has not been uploaded before.")
           MAX_POLLING_RETRIES.times do
-            if status == "SUCCESS"
+            if upload_status_response.success?
               UI.message("Uploaded APK/IPA Successfully!")
               break
-            elsif status == "IN_PROGRESS"
+            elsif upload_status_response.in_progress?
               sleep(POLLING_INTERVAL_SECONDS)
             else
               UI.message("Uploading the APK/IPA.")
               upload_binary(app_id, binary_path)
             end
-            status = upload_status(app_id, upload_token)
+            upload_status_response = get_upload_status(app_id, upload_token)
           end
-          if status != "SUCCESS"
-            UI.message("It took longer than expected to process your APK/IPA, please try again")
+          unless upload_status_response.success?
+            UI.message("It took longer than expected to process your APK/IPA, please try again.")
           end
         end
+        upload_status_response.release_id
       end
 
-      def self.upload_status(app_id, app_token)
+      # Gets the upload status for the app release
+      #
+      # Returns the status of the release. On success the release id exists and is nil in all other cases.
```

Comment:
This is not true, it actually crashes if the status is not found, and never returns nil. Right?

Reply:
No if the status is not found it should not crash since this is just checking the status of the upload. So if the upload does not exist this will let us know if it is in progress, does not exist, or if it is successful. 

I have included a screenshot that prints out the body of the response in upload status for uploading a new apk. At first it is error since it hasn't been uploaded before, then inprogress, then succeeds at the third check. Only then is there a release id. 
![Screen Shot 2020-07-01 at 10 45 12 AM](https://user-images.githubusercontent.com/32231345/86257994-4a9d7180-bb88-11ea-9981-74e1ccdcb39a.png)